src/coreclr/src/vm/zapsig.cpp

```diff
@@ -677,16 +685,19 @@ Module *ZapSig::DecodeModuleFromIndexIfLoaded(Module *fromModule,
         pAssembly = fromModule->GetAssembly();
     else
     {
-        if (index < fromModule->GetAssemblyRefMax())
+        NativeImage *compositeImage = fromModule->GetCompositeNativeImage();
+        uint32_t assemblyRefMax = (compositeImage != NULL ? 1 : fromModule->GetAssemblyRefMax());
```

Comment:
>1 [](start = 60, length = 1)

This isn't the same as the algorithm above in DecodeModuleFromIndex. (And this path is, I believe very rarely taken for R2R, so testing to success here is likely to be hard.)

Reply:
That's exactly why I missed that. I have modified it to use basically the same code as DecodeModuleFromIndex.

Reply:
Fixed in 11th commit.